pet-store-sql-repository/src/test/kotlin/com/johnkruss/osn/sql/GoodPetSqlRepositoryTest.kt

```diff
@@ -0,0 +1,115 @@
+package com.johnkruss.osn.sql
+
+import com.johnkruss.osn.generatePet
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.FreeSpec
+import io.kotest.matchers.shouldBe
+import org.jooq.exception.DataAccessException
+import org.jooq.exception.NoDataFoundException
+import java.util.UUID
+
+/*
+* Call this test whatever you want, I call it "proof your queries actually work"
+* When we extract our code out into small working pieces like this we don't even need a full application framework to
+* be able to run our tests. We can now effectively test this code in isolation! Of course this approach is still
+* compatible with a larger application framework, but it isn't required
+*
+* A quick note: Don't forget to run docker compose up -d over in your terminal before trying to run this test. We need
+* real database containers running!
+*
+* Goofed up your flyway migration and need to fix it? Good thing we tested here instead of against a live db right?
+* Simply spin down your docker containers and bring them back up. Now try again!
+*
+* Another quick note: I find it easier to run containers without mounted data volumes so that every time they're spun
+* down and back up, all the data in them is wiped out
+* */
+class GoodPetSqlRepositoryTest : FreeSpec({
+
+    lateinit var repository: PetSqlRepository
+    lateinit var dataSource: HikariDataSource
+
+    beforeSpec {
+        // Let's make a real connection, this could be parameterized for environment vars if needed
+        val config =
+            HikariConfig().apply {
+                jdbcUrl = "jdbc:postgresql://localhost:5432/pet_Store"
+                username = "username"
+                password = "password"
+            }
+        dataSource = HikariDataSource(config)
+
+        /*
+         * Look here, now we're testing our ACTUAL database migration code against an ACTUAL database!
+         * */
+        FlywayMigrator(dataSource).run()
+
+        repository = PetSqlRepository(dataSource)
+    }
+
+    afterSpec {
+        dataSource.close()
+    }
+
+    "Add a pet, then call it back" {
+        /*
+         * Note that we need randomness here because we're working against a live db
+         * If we used the same static strings we'd run into all sorts of wacky problems from duplicate data
```

Comment:
Maybe worth noting that this is just one way of doing it. Some engineers like avoiding randomness, some will create an ID that gets used by every test but then have a test cleanup in afterTest that wipes out the created rows. FWIW I prefer the random IDs for every test